### PR TITLE
Robustify workflow around testing defaults

### PR DESCRIPTION
**Pull request by mattpap**
This PR:

- removes `js-test` job from `bokeh-ci` workflow
- makes `bokehjs-ci` workflow run always
- commits `defaults.yaml` to the repository
  - this works in a similar way to bokehjs' baselines
  -  ~240 kb; could be much less if inherited yet unchanged properties were filtered out
- adds `tests/test_defaults` to collect defaults and diff with committed baseline
- makes bokehjs' tests use committed defaults for testing

This allows simplify the overall workflow, removes duplicate efforts (`js-test`) and allows to seal in the defaults, allowing comparison between revisions and between a revision and current state.

~~The baseline is stored in a JSON file (`tests/baselines/defaults.json`), which isn't an ideal format, due to lack of trailing commas, which makes diffing less effective as it could have been otherwise. Perhaps using JSON5 would be a simple way to improve this in the future~~.

### CODE REVIEW

File: tests/test_defaults.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+#-----------------------------------------------------------------------------
+
+class TestDefaults:
+    def test_defaults(self) -> None:
+        baseline = Path(__file__).parent / "baselines" / "defaults.json"
+
+        defaults = collect_defaults()
+        output_defaults(baseline, defaults)
+
+        status, out, _ = diff_baseline(baseline)
```
Comment (bryevdv) [inline tests/test_defaults.py:41]: What is the process to get test passing again on occasions where we do intentionally have to change a default value?

**Comment by bryevdv** [comment]
> ~300 kb; could be much less if inherited yet unchanged properties were filtered out
Maybe the time has finally come we should consider a sub-repo for all these large artifact collections.

File: tests/test_defaults.py (hunk 1 above)
Comment (mattpap) [reply tests/test_defaults.py:41]: The procedure is the same as in bokehjs' tests. The test generates a new baseline and overwrites the existing one in the repository. If there's a difference, the test fails and the diff is showed. The same diff can be obtained with `git diff`. Differences are reviewed and, if everything is in order, committed to the repository.

**Comment by mattpap** [comment]
> Maybe the time has finally come we should consider a sub-repo for all these large artifact collections.

Maybe at some point. I want to significantly expand testing of generated JSON (in tests and examples) and test it in bokehjs, so that could tip the scale. For reference, during the past 12 months, the repository gained only 17 MB from all changes to bokehjs' `*.png` linux baseline files. For me it's only, given the extent of changes for 3.0.

**Comment by bryevdv** [comment]
@mattpap this should have had more work/discussion before merging: 

* Workflow details, file locations, how to edit, etc. need to be explicitly drawn out in the dev guide for others
* Lots of overlap with `scripts/spec.py` it might be better to have a runnable script to generate these defaults so that they can be generated and archived for each release, etc.

**Comment by bryevdv** [comment]
Another reason this would have benefitted from more consideration: it's actually the ref guide generation that takes several minutes to finish. Had this been done with that also in mind, we might have had a basis to substantially speed up docs builds. A JSON representation for defaults with a sidecar file for helpstrings could eliminate most of the work bokeh model and prop directives do.

**Pull request merged**